compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/WindowInfo.skiko.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.platform
+
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.input.pointer.EmptyPointerKeyboardModifiers
+import androidx.compose.ui.input.pointer.PointerKeyboardModifiers
+import androidx.compose.ui.unit.IntSize
+
+/**
+ * Provides information about the Window that is hosting this compose hierarchy.
+ */
+@Stable
+actual interface WindowInfo {
+    /**
+     * Indicates whether the window hosting this compose hierarchy is in focus.
+     *
+     * When there are multiple windows visible, either in a multi-window environment or if a
+     * popup or dialog is visible, this property can be used to determine if the current window
+     * is in focus.
+     */
+    actual val isWindowFocused: Boolean
+
+    /**
+     * Indicates the state of keyboard modifiers (pressed or not).
+     */
+    @ExperimentalComposeUiApi
+    actual val keyboardModifiers: PointerKeyboardModifiers
+
+    @ExperimentalComposeUiApi
+    val size: IntSize
+}
+
+
+internal class WindowInfoImpl : WindowInfo {
+    private val _isWindowFocused = mutableStateOf(false)
+
+    override var isWindowFocused: Boolean
+        set(value) { _isWindowFocused.value = value }
+        get() = _isWindowFocused.value
+
+    @ExperimentalComposeUiApi
+    override var keyboardModifiers: PointerKeyboardModifiers
+        get() = GlobalKeyboardModifiers.value
+        set(value) { GlobalKeyboardModifiers.value = value }
+
+    @ExperimentalComposeUiApi
+    override var size: IntSize = IntSize.Zero
```

Comment:
Shouldn't this be a "State"?

Reply:
No

Reply:
Why not? The other properties are state. How can the user get notified about changes?

Reply:
You probably right 🤔, I had an impression that `Local*.current` does this job

Reply:
> You probably right 🤔, I had an impression that Local*.current does this job

It does, but only if you replace the whole (immutable) object when something changes, like (for example) in `LocalTextStyle`.